### DISCUSSION TITLE
Bugs with Custom Theming

### DISCUSSION THREAD

**Discussion by oklopfer** [isontheline/pro.webssh.net#1171] [Q&A]
Just want to start by saying I have been looking for an iOS SSH client that allows both custom ANSI themes + custom system fonts, and your app is the *only* one I could find - a be all, end all! I think this feature should be highlighted more.

That being said, I have found a few bugs with the custom ANSI themes:
1. No matter the color they are set to in a theme, and no matter what theme you choose, Purple and Bright Purple are stuck being the same color scheme as the WebSSH theme. It is pretty easy to see the issue with themes like Desert or Seashells, which intentionally have no purples.

2. In creating a custom theme, I cannot change the Cursor, background, foreground, and selection colors. I have tried everything I can to alter them in all of the tabs (Classic, Grid, RGB), but they just won't budge. I had to find a theme that had close enough values to what I wanted, clone it, and then edit the "other colors" (ansi codes).

3. While I was successful in setting my other ANSI color codes, notably, the hex field in RGB refuses to set the color to that value. When exiting the screen, no changes are ever applied. I had to enter the RGB-256 values to set them correctly.

Thank you again for your app and time!

**Comment by isontheline** [comment]
Hello @oklopfer 👋

Thank you so much for your great and strong feedback about WebSSH 🙏

> 1. No matter the color they are set to in a theme, and no matter what theme you choose, Purple and Bright Purple are stuck being the same color scheme as the WebSSH theme. It is pretty easy to see the issue with themes like Desert or Seashells, which intentionally have no purples.

Very good catch!
I’m using “purple” and “brightPurple” instead of “magenta” and “brightMagenta” :(
I will fix it asap in the next upcoming minor release.

> 2. In creating a custom theme, I cannot change the Cursor, background, foreground, and selection colors. I have tried everything I can to alter them in all of the tabs (Classic, Grid, RGB), but they just won't budge. I had to find a theme that had close enough values to what I wanted, clone it, and then edit the "other colors" (ansi codes).

Oops seems to be a regression, I’m looking at it

> 3. While I was successful in setting my other ANSI color codes, notably, the hex field in RGB refuses to set the color to that value. When exiting the screen, no changes are ever applied. I had to enter the RGB-256 values to set them correctly.

That’s strange I’ve just tested over iOS 17 and macOS 14 and I can apply hex values. The color is even updated when I focus another field.

Sorry @oklopfer for all these troubles
I’m opening an issue to track them all : https://github.com/isontheline/pro.webssh.net/issues/1173
